In [1]:
import glob
import time
from datetime import datetime

import os
from os.path import exists

import requests
from bs4 import BeautifulSoup

import re

In [3]:
url = 'https://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-Chicago/'

html_food = requests.get(url).text

soup_food= BeautifulSoup(html_food, "lxml")
soup_food.title

cafe=[];rank=[];food=[];link=[];
price=[];add=[];num=[];page=[];

s = soup_food.select('div.sammyListing')

for i in range(0, len(s)):

    rank.append(soup_food.select('div.sammyRank')[i].get_text())
    food.append(soup_food.select("b")[i].get_text())
    cafe.append(soup_food.select('div.sammyListing')[i].get_text().split('\n')[1])
    link.append('https://www.chicagomag.com{}'.format(soup_food.find_all('div','sammy')[i].find('a')["href"]))
    
    print('https://www.chicagomag.com{}'.format(soup_food.find_all('div','sammy')[i].find('a')["href"]))
                                                                                                            
    hf= requests.get(link[i]).text
    sf= BeautifulSoup(hf,"lxml")
    
    b=sf.select("p.addy em")[0].get_text()
    
    price.append(b.split('. ')[0])
    add.append(b[len(b.split('. ')[0])+2:re.compile('\d+[-]\d+[-]\d+').search(b).start()-2])
    page.append(b.split(',')[-1])
    num.append(re.compile('\d+[-]\d+[-]\d+').search(b).group())

https://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Old-Oak-Tap-BLT/
https://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Au-Cheval-Fried-Bologna/
https://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Xoco-Woodland-Mushroom/
https://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Als-Deli-Roast-Beef/
https://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Publican-Quality-Meats-PB-L/
https://www.chicagomag.comhttp://www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Hendrickx-Belgian-Bread-Crafter-Belgian-Chicken-Curry-Salad/


ConnectionError: HTTPSConnectionPool(host='www.chicagomag.comhttp', port=443): Max retries exceeded with url: //www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Hendrickx-Belgian-Bread-Crafter-Belgian-Chicken-Curry-Salad/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00000223209BE9B0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))

###ConnectionError: HTTPSConnectionPool(host='www.chicagomag.comhttp', port=443): Max retries exceeded with url: //www.chicagomag.com/Chicago-Magazine/November-2012/Best-Sandwiches-in-Chicago-Hendrickx-Belgian-Bread-Crafter-Belgian-Chicken-Curry-Salad/ (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x00000223209BE9B0>: Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))



In [ ]:
이런 에러가 떠서 저장은 못했습니다